src/core/credentials/transport/alts/grpc_alts_credentials_options.h

```diff
@@ -50,6 +70,7 @@ typedef struct target_service_account {
 typedef struct grpc_alts_credentials_client_options {
   grpc_alts_credentials_options base;
   target_service_account* target_account_list_head;
+  std::shared_ptr<grpc::alts::TokenFetcher> token_fetcher = nullptr;
```

Comment:
Please make sure that we are creating and destroying `grpc_alts_credentials_client_options` via C++ `new` and `delete` rather than C-style `malloc()` and `free()`.  This is necessary for any struct with a C++ type as a data member, because otherwise we won't properly call the ctor and dtor of the data member when the outer struct is created and destroyed.

Reply:
Looks like the entire stack is written using `gpr_zalloc` (`malloc` behind the scene), so if I switch to `delete` here a lot of will have to be rewritten, which I'd like to defer to a separate PR.

Reply:
I think this needs to be done as part of this PR.  The code as currently written is not safe, because we have no guarantee that it's safe to use a type that we don't own like `std::shared_ptr<>` without actually calling its ctor.